pkg/asset/store.go

```diff
@@ -14,47 +20,212 @@ type Store interface {
 	Fetch(Asset) error
 }
 
+// FileFetcher returns a list of files which satisfy the condition.
+// The input must be either a compiled regular expression (*regexp.Regexp),
```

Comment:
@staebler how can you determine when to use the filefetcher or patternfetcher in the Fetch()?

Reply:
The fetcher interface would contain both functions. The asset would determine which function to call from the fetcher, in the same way that it does now by passing in different types.

```
type FileFetcher interface {
  Get(string) (*File, error)
  GetByPattern(*regexp.Regexp) ([]*File, error)
}
```

In, for example, bootstrap's Load,
```
func (a *Bootstrap) Load(p asset.FileFetcher) (found bool, err error) {
  a.file, err = p.Get(bootstrapIgnFilename)
  return a.file != nil, err
}
```

Reply:
The asset store would hold a `FileFetcher` instead of `onDiskAssets`, and pass that `FileFetcher` to the assets' `Load` methods.

```
type fileFetcher struct {
  onDiskAssets map[string][]byte
}

func (f *fileFetcher) Get(string) (File, error) {...}

func (f *fileFetcher) GetByPattern(*regexp.Regexp) (File, error) {...}
```